prov/verbs/src/verbs_info.c

```diff
@@ -897,8 +897,10 @@ static int verbs_devs_add(struct dlist_entry *verbs_devs, char *dev_name,
 	addr->rai = rai;
 
 	dlist_foreach_container(verbs_devs, struct verbs_dev_info, dev, entry)
-		if (!strcmp(dev_name, dev->name))
+		if (!strcmp(dev_name, dev->name)) {
+			free(dev_name);
 			goto add_rai;
+		}
```

Comment:
The structure of the verbs code to get to this path leaves a lot to be desired.  The fix is good, and I don't see a simpler way of handling this without some sort of major rework of the code.  Thanks!

Reply:
I was having the same feeling. One semantic is to say at `verbs_devs_add` we are transferring the ownership for both `dev_name` and `rai` -- failure or success. Then we should move the failure case `free` from https://github.com/hzhou/libfabric/blob/dfb615a6a3ab2deaaf57934a26b2a84581e720b6/prov/verbs/src/verbs_info.c#L1048-L1049 into `verbs_devs_add`. I think that'll make the code cleaner.

Reply:
I could go with that model